cluster-provision/okd/scripts/run.sh

```diff
@@ -95,10 +139,42 @@ if [[ ${worker_node_ip} != "192.168.126.51" ]]; then
     haproxy -f /etc/haproxy/haproxy.cfg
 fi
 
-until [[ $(oc get pods --all-namespaces --no-headers | grep -v revision-pruner | grep -v Running | grep -v Completed | wc -l) -le 3 ]]; do
-    echo "waiting for pods to come online"
+set +xe
+n=0
+while true; do
```

Comment:
to be honest it hard to understand what happens under this loop. can you please add comment what the whole loop should do, why do you need to `cat /tmp/num_pods.txt`, why do you need to inject timeout